nomad/rpc.go

```diff
@@ -126,7 +154,37 @@ func (s *Server) handleConn(conn net.Conn, isTLS bool) {
 			return
 		}
 		conn = tls.Server(conn, s.rpcTLS)
-		s.handleConn(conn, true)
+
+		// Force a handshake so we can get information about the TLS connection
+		// state.
+		tlsConn, ok := conn.(*tls.Conn)
+		if !ok {
+			s.logger.Printf("[ERR] nomad.rpc: expected TLS connection but got %T", conn)
+			conn.Close()
+			return
+		}
+
+		if err := tlsConn.Handshake(); err != nil {
+			s.logger.Printf("[WARN] nomad.rpc: failed TLS handshake from connection from %v: %v", tlsConn.RemoteAddr(), err)
+			conn.Close()
+			return
+		}
+
+		// Update the connection context with the fact that the connection is
+		// using TLS
+		ctx.TLS = true
+
+		// Parse the region and role from the TLS certificate
+		state := tlsConn.ConnectionState()
+		parts := strings.SplitN(state.ServerName, ".", 3)
```

Comment:
Is this name guarenteed to be in this format always?

Reply:
It should be in the case of verifying hostnames but it is a bit brittle. Going to rework

Reply:
See newest commit. I store the whole verified chain so we can work with the certificates where needed.